File: 1752_Check_if_Array_Is_Sorted_and_Rotated/solution.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        for i in range(n):
            f = []
            for j in range(i, n):
                f.append(nums[j])
            for j in range(i):
                f.append(nums[j])
            s = True
            for j in range(n - 1):
                if f[j] > f[j + 1]:
                    s = False
                    break
            if s:
                return True
        return False

File: 1752_Check_if_Array_Is_Sorted_and_Rotated/solution.c
bool check(int* nums, int n) {
    for (int i = 0; i < n; i++) {
        int* f = (int*)malloc(n * sizeof(int));
        int idx = 0;
        for (int j = i; j < n; j++) f[idx++] = nums[j];
        for (int j = 0; j < i; j++) f[idx++] = nums[j];
        bool s = true;
        for (int j = 0; j < n - 1; j++) if (f[j] > f[j + 1]) { s = false; break; }
        free(f);
        if (s) return true;
    }
    return false;
}
